src/cryocanvas/app.py

```diff
@@ -128,19 +131,22 @@ def _add_widget(self):
         self._connect_events()
 
     def _connect_events(self):
-        # Use a partial function to pass additional arguments to the event handler
-        on_data_change_handler = tz.curry(self.on_data_change)(app=self)
         for listener in [
             self.viewer.camera.events,
             self.viewer.dims.events,
-            self.painting_layer.events.paint,
         ]:
             listener.connect(
                 debounced(
-                    ensure_main_thread(on_data_change_handler),
+                    self.on_view_change,
```

Comment:
I think we can just use the bound instance method instead of currying or using a partial.